rero_ils/modules/holdings/api_views.py

```diff
@@ -139,3 +149,98 @@ def receive_regular_issue(holding_pid):
         abort(400)
     # the created item of type issue is returned
     return jsonify({'issue': issue})
+
+
+def do_holding_jsonify_action(func):
+    """Jsonify loan actions for holdings methods.
+
+    This method for the circulation actions that required access to the holding
+    object before executing the invenio-circulation logic.
+    """
+    @wraps(func)
+    def decorated_view(*args, **kwargs):
+        try:
+            data = flask_request.get_json()
+            if not data.get('description'):
+                raise MissingRequiredParameterError
+            holding_pid = data.pop('holding_pid', None)
+            holding = Holding.get_record_by_pid(holding_pid)
+
+            if not holding:
+                abort(404)
+
+            # create a provisional item
+            item_metadata = {
+                'type': 'provisional',
+                'document': {
+                    '$ref': get_ref_for_pid('doc', holding.document_pid)},
+                'location': {
+                    '$ref': get_ref_for_pid('loc', holding.location_pid)},
+                'item_type': {'$ref': get_ref_for_pid(
+                    'itty', holding.circulation_category_pid)},
+                'enumerationAndChronology': data.get('description'),
+                'status': ItemStatus.ON_SHELF,
+                'holding': {'$ref': get_ref_for_pid('hold', holding.pid)}
+            }
+            item = Item.create(item_metadata, dbcommit=True, reindex=True)
+            data.pop('description', None)
+            item_data, action_applied = \
```

Comment:
`item_data` seems not used, you could write : 
```python
   _, action_applied = func(holding, item, data, *args, **kwargs)
```
Better for memory usage and garbage collector